### ISSUE TITLE
Make Template Details View a Modal

### ISSUE THREAD

**Issue by BrittanyIRL** [swissspidy/web-stories-wp#9671]
## Feature Description

This came up in a design sync (11/9/2021) between @agingoldseco and pea pod (@littlemilkstudio, @samwhale, @BrookeGraham , @BrittanyIRL). The template detail view (see screenshot) was originally styled to look like a modal but because of really heavy assets (loading each story page as a preview) it was done as stand alone pages. This means every time you click "see details" on the explore template page of the dashboard you're navigating away from explore templates to this specific template view. That's fine. As of #9508 where we introduce searching templates, this navigation gets more noticeable since you can now search and see details and then when you click that 'x' you expect your search results but instead your search term is completely gone and you're back at the top of the page. 

We've got a couple options (probably more that I'm unaware of - this is just what we briefly discussed): 
- Update the 'X' button to say 'back' - so that users can glean that the view was updated.
- Make the detail view a modal. We fixed the perf issues with the page load so now a modal actually is doable. 

This is potentially a good roadmap item for 1.16 

![Screen Shot 2021-11-10 at 8 31 50 AM](https://user-images.githubusercontent.com/10720454/141142633-3754c83e-24cf-4dda-ac2d-40396a7ff128.png)

(With content below fold)
![Screen Shot 2021-11-10 at 8 30 52 AM](https://user-images.githubusercontent.com/10720454/141142514-d008a93c-4559-467b-8f0e-594a302f8aa5.png)

**Comment by gajasinski** [comment]
@BrittanyIRL can you please attach the screen shot?

**Comment by BrittanyIRL** [comment]
oh whoops! yup on it.

**Comment by dreamofabear** [comment]
> you can now search and see details and then when you click that 'x' you expect your search results but instead your search term is completely gone and you're back at the top of the page.

Supporting a way to view a template's details and preserve search results makes a lot of sense to me.

**Comment by agingoldseco** [comment]
My thoughts on this, also discussed with Brittany:

- It should definitely be/act like a modal—clicking X should always go back to search results if there was a search
- Browser back button can won't go back to search, but previous page—in the current interface, it will always go back to the main template explore page
- Arrows on either side of the modal should also reflect search results
- Right arrow should gray out if user reaches the end of search results 

I think those would be good, simple enough changes to transition users into being able to search now, before doing any majorly visible overhauls. Please let me know if you notice anything I'm not taking into account!

**Comment by swissspidy** [comment]
This was also just mentioned to me by rtCamp engineers:

> [...] The single template view looks like popup layout and can be closed by clicking on X icon. [...] Can we add ESC key support to go back.

So yeah, if we get ESC support by switching to a modal, that would be great.

**Comment by BrookeGraham** [comment]
@agingoldseco I ripped out the related templates. Now just waiting on close button placement, font size, and white space around heading.
re:
https://formidablelabs.slack.com/archives/C01ESBB8ZN1/p1638456480017900

**Comment by kkalarickal** [comment]
Confirmed on https://stories-qa-wordpress-amp.pantheonsite.io/ using PR #9875 

Status: LGTM 👍 

Scenarios:
1. Clicking browser back button 
- after opening modal via "See details", user is taken to last URL that was used before accessing the Explore templates (this could be the editor or the dashboard) ✔️ 
- after completing a search, user is taken to last URL even from the search results page ✔️ 
- same back button behavior after user navigates between templates in the explore templates or search results ✔️ 

2. Clicking the X button on modal or typing ESC key
- user is taken back to explore template or search results ✔️ 
- user is taken back to search results even after navigating using forward/back arrows ✔️ 

3. Right and Left arrows are greyed out within search results if user is viewing details of the last or first result template ✔️

**Comment by kkalarickal** [comment]
@swissspidy - After opening template details modal,
-  if user chooses to "Use template" to open the editor, and then clicks back button, she is brought back to the search result/explore template. 
- if the user uses template, then clicks back to explore templates, and tries to use back button ... the search page is no longer part of browser history.
